pandas/core/arrays/categorical.py

```diff
@@ -1732,17 +1744,25 @@ def fillna(self, value=None, method=None, limit=None):
         return self._constructor(values, categories=self.categories,
                                  ordered=self.ordered, fastpath=True)
 
-    def take_nd(self, indexer, allow_fill=True, fill_value=None):
-        """ Take the codes by the indexer, fill with the fill_value.
+    def take_nd(self, indexer, allow_fill=None, fill_value=None):
+        """
+        Return the indices
 
         For internal compatibility with numpy arrays.
         """
+        indexer = np.asarray(indexer, dtype=np.intp)
+        if allow_fill is None:
+            if (indexer < 0).any():
+                warn(_take_msg, FutureWarning, stacklevel=2)
+                allow_fill = True
 
-        # filling must always be None/nan here
-        # but is passed thru internally
-        assert isna(fill_value)
+        if fill_value is None or isna(fill_value):
+            # For backwards compatability, we have to override
+            # any na values for `fill_value`
```

Comment:
I don't understand this comment

Reply:
But `np.nan` is the NA value for Categorical, so shouldn't its `take` method be able to check that anyhow (regardless from back compat reasons) ?